File: src/CNAB.Importer.API/Infrastructure/Data/Entities/Transaction.cs
namespace CNAB.Importer.API.Infrastructure.Data.Entities;

public class Transaction : BaseEntity
{
    public TTransaction Type { get; set; }
    public DateTime Date { get; set; }
    public decimal Value { get; set; }
    public string Document { get; set; }
    public string Card { get; set; }
    public string Hour { get; set; }
    public string StoreOwnerName { get; set; }
    public string StoreName { get; set; }
}

public enum TTransaction : byte
{
    Debit = 1,
    Boleto = 2,
    Financing = 3,
    Credit = 4,
    LoanReceipt = 5,
    Sales = 6,
    TedReceipt = 7,
    DocReceipt = 8,
    Rent = 9
}

File: src/CNAB.Importer.Client/src/models/transaction.ts
export interface StoreTransaction {
  storeName: string;
  totalAmount: number;
  transactions: Transaction[];
}

export interface Transaction {
  type: TransactionType;
  date: Date;
  value: number;
  document: string;
  card: string;
  hour: string;
  storeOwnerName: string;
  storeName: string;
}

export enum TransactionType {
  debit = 1,
  boleto = 2,
  financing = 3,
  credit = 4,
  loanReceipt = 5,
  sales = 6,
  tedReceipt = 7,
  docReceipt = 8,
  rent = 9,
}
